src/components/SubNav/SubNav.tsx

```diff
@@ -0,0 +1,318 @@
+                );
+                return null;
+              }
+            }
+          );
+        }
+      };
+
+      // Validate primaryActions and secondaryActions
+      validateActions(primaryActions, "primaryActions");
+      validateActions(secondaryActions, "secondaryActions");
+
+      const backgroundColor =
+        highlightColor !== undefined ? actionBackgroundColor : undefined;
+
+      const styles = useMultiStyleConfig("SubNav", {
+        backgroundColor: backgroundColor,
+        highlightColor: highlightColor,
+      });
+
+      const handleScroll = useCallback(() => {
+        if (scrollableRef.current) {
+          const { scrollLeft, scrollWidth, clientWidth } =
+            scrollableRef.current;
+
+          // Determine if we are scrolling to the left or right
+          const isScrollingRight = scrollLeft > lastScrollLeft;
+          setLastScrollLeft(scrollLeft); // Update last scroll position
+
+          // Show right fade only if scrolling to the left
+          if (!isScrollingRight) {
+            // Show right fade only if not at the end
+            const atRightEdge = scrollLeft + clientWidth >= scrollWidth;
+            setShowRightFade(!atRightEdge);
+          } else {
+            setShowRightFade(false); // Hide when scrolling right
+          }
+        }
+      }, [lastScrollLeft, setLastScrollLeft, setShowRightFade]);
+
+      useEffect(() => {
+        const refCurrent = scrollableRef.current;
+        if (refCurrent) {
+          refCurrent.addEventListener("scroll", handleScroll);
+
+          // Initial check to set the right fade effect based on the initial scroll state
+          const { scrollWidth, clientWidth, scrollLeft } = refCurrent;
+          const atRightEdge = scrollLeft + clientWidth >= scrollWidth;
+
+          // Set the right fade effect if content is scrollable and not at the right edge
+          setShowRightFade(!atRightEdge);
+
+          // If the scrollable area is already at the end, hide the right fade initially
+          if (scrollWidth <= clientWidth) {
+            setShowRightFade(false); // If the content fits in the viewport, no need for a fade
+          }
+        }
+
+        // Cleanup event listener on component unmount
+        return () => {
+          if (refCurrent) {
+            refCurrent.removeEventListener("scroll", handleScroll);
+          }
+        };
+      }, [handleScroll]);
+
+      return (
+        <Box __css={styles.base}>
+          <Flex
+            alignItems="baseline"
+            className={className}
+            gap="1rem"
+            justify="space-between"
+          >
+            <List
+              type="ul"
+              m="0"
+              sx={{ ...styles.scrollableButtons, ...styles.primaryActions }}
+              ref={scrollableRef}
+              inline
+              noStyling
+            >
+              <li id="primary-actions">{primaryActions}</li>
```

Comment:
I feel like each button/link should be an li from an accessibility standpoint rather than putting the whole group of buttons/links in one li.